### PR TITLE
fix(schema): support draft 6 and 7

### PR DESCRIPTION
**Pull request by P0lip**
Fixes https://github.com/stoplightio/spectral/issues/573

**Checklist**

- [x] Tests added / updated
- [x] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

**Additional context**

Contains 2 unrelated fixes (see commit messages).

Regarding ajv-better-errors and the exception that's thrown - I'll try to provide a fix for the package.

This is where it fails:
https://github.com/atlassian/better-ajv-errors/blob/126ee2e45ee829e69e3b2f3f6d56eb49d5c42135/src/validation-errors/enum.js#L52
https://github.com/janl/node-jsonpointer/blob/master/jsonpointer.js#L60
Basically, jsonpointer does not expect non-object value.

### CODE REVIEW

File: test-harness/scenarios/external-schemas-ruleset.scenario [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+info:
+  title: foo
+  description: baz
+====command====
+lint {document} --ruleset ./test-harness/scenarios/rulesets/external-schemas-ruleset.yaml
+====stdout====
+OpenAPI 3.x detected
+
+{document}
+ 3:10  error  info-title        should be equal to one of the allowed values
```
Comment (philsturgeon) [inline test-harness/scenarios/external-schemas-ruleset.scenario]: Could you make a feature request for adding enum values to the message? This message looks funny.
